lib/tessel/deployment/rust.js

```diff
@@ -33,32 +37,134 @@ var exportables = {
         program
       };
     },
-    shell: () => {
+    shell: (opts) => {
       return tags.stripIndent `
         #!/bin/sh
-        exec /app/remote-script/rust_executable
+        exec /app/remote-script/${opts.resolvedEntryPoint}
```

Comment:
This is related to https://github.com/tessel/t2-cli/issues/785 and will have to be fixed when that linked issue is. Basically, we untar to `/remote-script` whether we `push` or `run` so `/app/remote-script` gets created.